cmd/gateway-common.go

```diff
@@ -287,6 +289,28 @@ func ToMinioClientCompleteParts(parts []CompletePart) []minio.CompletePart {
 	return mparts
 }
 
+// IsBackendOnline - verifies if the backend is reachable
+// by performing a GET request on the URL. returns 'true'
+// if backend is reachable.
+func IsBackendOnline(ctx context.Context, clnt *http.Client, urlStr string) bool {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+	defer cancel()
+
+	// never follow redirects
+	clnt.CheckRedirect = func(*http.Request, []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, urlStr, nil)
+	if err != nil {
+		return err == nil
+	}
+	if _, err = clnt.Do(req); err != nil {
+		return xnet.IsNetworkOrHostDown(err)
```

Comment:
xnet.IsNetworkOrHostDown(err) can return 'true', although this function name is IsBackendOnline()

Reply:
Oh yes my bad needs to be fixed..